app/bundles/EmailBundle/Model/EmailModel.php

```diff
@@ -2314,4 +2315,22 @@ public function buildUrl($route, $routeParams = [], $absolute = true, $clickthro
 
         return $url;
     }
+
+    public function getPublishStatus(Email $email): string
+    {
+        $publishStatus = $email->getPublishStatus();
+        if ($email->isSegmentEmail() && $email->getPublishUp()) {
+            if ('published' == $publishStatus) {
+                if ($email->isContinueSending()) {
+                    $publishStatus = 'running';
+                } elseif ($email->getPendingCount()) {
+                    $publishStatus = 'running';
+                } else {
+                    $publishStatus = 'sent';
+                }
+            }
+        }
+
+        return $publishStatus;
+    }
```

Comment:
My suggestions:
1. Move this method to the Email entity itself.
2. Rewrite it to [return early](https://dev.to/jpswade/return-early-12o5) as this is hard to reason about.
3. Use constants for the statuses instead of repeating strings.